core/vm/actor/builtin/types/market/deal_info_manager/impl/deal_info_manager_impl.cpp

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "deal_info_manager_impl.hpp"
+
+#include "vm/actor/builtin/types/market/publish_deals_result.hpp"
+
+namespace fc::vm::actor::builtin::types::market::deal_info_manager {
+  using v0::market::PublishStorageDeals;
+
+  outcome::result<CurrentDealInfo> DealInfoManagerImpl::getCurrentDealInfo(
+      const DealProposal &proposal, const CID &publish_cid) {
+    // TODO (ortyomka): maybe async call, it's long
```

Comment:
Can we get callback via parameter?

Reply:
No, we have to wait for result to use in caller (sealing).